src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -1063,6 +1063,46 @@ public static String convertUnderscoreToSpace(@Nonnull CharSequence sequence) {
         return UNDERSCORE_TO_SPACE.matcher(sequence).replaceAll(" ");
     }
 
+    /**
+     * Tries to parse a string into an int, returning a default value if it fails.
+     * @param val string to parse
+     * @param defaultValue default value to return
+     * @return returns an int from the parsed string, otherwise the default value
+     */
+    public static int tryParseInt(String val, int defaultValue){
+        try {
+            return Integer.parseInt(val);
+        } catch (NumberFormatException e) {
+            GTLog.logger.warn(e);
+        }
+        return defaultValue;
+    }
+
+    /**
+     * Compares a value against a min and max, with an option to invert the logic
+     * @param value value to be compared
+     * @param maxValue the max that the value can be
+     * @param minValue the min that the value can be
+     * @param isInverted whether to invert the logic of this method
+     * @return an int from 0 (value <= min) to 15 (value >= max) normally, with a ratio when the value is between min and max
+     */
+    public static int compareValue(int value, float maxValue, float minValue, boolean isInverted) {
```

Comment:
This method should be renamed to make it clear that this is for redstone signal strength.

Reply:
resolved with https://github.com/GregTechCEu/GregTech/pull/1448/commits/c266abbb0104c2e5c5994051ad4c881576623756